process/interceptors/factory/interceptedMetaHeaderDataFactory.go

```diff
@@ -58,6 +65,9 @@ func NewInterceptedMetaHeaderDataFactory(argument *ArgInterceptedDataFactory) (*
 	if check.IfNil(argument.ValidityAttester) {
 		return nil, process.ErrNilValidityAttester
 	}
+	if check.IfNil(argument.ProofsPool) {
```

Comment:
not need, it is only forwarded to the component

Reply:
removed check